### PR TITLE
feat(console): add interactionMode qs to AC

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

Allow AC to use the signup first user experience by a given interaction mode query params. 

e.g.  http://localhost:3002/console?signup=true

- bump the @logto/react SDK version
- read queryString `signup` from the search params. If the signup mode is specified, should use signup first user experience. 

## Testing

test locally

## Checklist

- [ ] `.changset-staged`
- [ ] unit tests
- [ ] integration tests
- [ ] docs

OR

- [x] This PR is not applicable for the checklist

### CODE REVIEW

File: packages/console/src/containers/AppContent/index.tsx [hunk 1]
```diff
@@ -29,16 +30,22 @@ const AppContent = () => {
...
   const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
 
   useEffect(() => {
     if (!isAuthenticated && !isLogtoLoading) {
-      void signIn(new URL(href, window.location.origin).toString());
+      const signUpFirst = searchParameters.get(queryParameterKeys.signUp);
+
+      void signIn(
+        new URL(href, window.location.origin).toString(),
+        conditional(signUpFirst === 'true' && 'signUp')
```
Comment (xiaoyijun) [inline packages/console/src/containers/AppContent/index.tsx:45]:
```suggestion
        conditional(yes(signUpFirst) && 'signUp')
```

**Review by xiaoyijun — APPROVED** [review]

File: packages/console/src/containers/AppContent/index.tsx [hunk 2]
```diff
@@ -29,16 +30,22 @@ const AppContent = () => {
...
+      );
     }
-  }, [href, isAuthenticated, isLogtoLoading, signIn]);
+  }, [href, isAuthenticated, isLogtoLoading, searchParameters, signIn]);
```
Comment (gao-sun) [inline packages/console/src/containers/AppContent/index.tsx:48]: a direct access should be sign-in? maybe we only need to set to sign-up when user comes from elsewhere like website and product hunt

Comment (gao-sun) [reply packages/console/src/containers/AppContent/index.tsx:48]: but for now we cannot configure to use sign-up as the first screen by default. keep as it is looks ok

**Review by gao-sun — APPROVED** [review]

File: packages/console/src/consts/index.ts [hunk 3]
```diff
@@ -10,3 +10,7 @@ export const appearanceModeStorageKey = 'logto:admin_console:appearance_mode';
 export const profileSocialLinkingKeyPrefix = 'logto:admin_console:linking_social_connector';
 export const requestTimeout = 20_000;
 export const defaultPageSize = 20;
+
+export enum queryParameterKeys {
```
Comment (gao-sun) [inline packages/console/src/consts/index.ts:14]: image

maybe we can align naming across packages. both are good to me

File: packages/console/src/containers/AppContent/index.tsx (hunk 2 above)
Comment (simeng-li) [reply packages/console/src/containers/AppContent/index.tsx:48]: The current use case only supports the Logto Official WebSite's button, assume all lands at the register page.

File: packages/console/src/consts/index.ts (hunk 3 above)
Comment (simeng-li) [reply packages/console/src/consts/index.ts:14]: cool updated

File: packages/console/src/containers/AppContent/index.tsx (hunk 2 above)
Comment (gao-sun) [reply packages/console/src/containers/AppContent/index.tsx]: yes, so keep as it is looks ok now

**Review by wangsijie — APPROVED** [review]

File: packages/console/src/consts/index.ts (hunk 3 above)
Comment (gao-sun) [reply packages/console/src/consts/index.ts:14]: should update ui as well?

Comment (simeng-li) [reply packages/console/src/consts/index.ts]: align with ui `searchKeys`

**Pull request merged**